contracts/libraries/WeightedOracleLibrary.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity >=0.5.0 <0.8.0;
+
+import '@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol';
+
+/// @title Weighted Oracle library
+/// @notice Provides functions to integrate with different tier oracles of the same V3 pair
+library WeightedOracleLibrary {
+
+    /// @notice Fetches liquidity-time-weighted average tick across different tiers of the same Uniswap V3 pair
+    /// @param pools Addresses of different tiers of the same Uniswap V3 pair that we want to observe
+    /// @param period Number of seconds in the past to start calculating liquidity-time-weighted average
+    /// @return liquidityTimeWeightedAverageTick The liquidity-time-weighted average tick from (block.timestamp - period) to block.timestamp
+    function consult(address[] calldata pools, uint32 period) internal view returns (int24 liquidityTimeWeightedAverageTick) {
+        require(period != 0, 'BP');
+
+        uint32[] memory secondsAgos = new uint32[](2);
+        secondsAgos[0] = period;
+        secondsAgos[1] = 0;
+        uint160 periodX128 = uint160(period) << 128;
```

Comment:
computing periodX128 and then passing into inner function is weird when you already pass period and this is a simple shift